src/main/java/org/openstreetmap/atlas/geography/atlas/dynamic/DynamicAtlas.java

```diff
@@ -801,6 +805,47 @@ else if (entity instanceof Relation)
                         entity.getClass().getName());
             }
         }
+        if (this.policy.isAggressivelyExploreRelations() && !loop)
+        {
+            // Get all the neighboring shards
+            final Set<Shard> onlyNeighboringShards = new HashSet<>();
+            this.loadedShards.keySet().forEach(
+                    shard -> this.sharding.neighbors(shard).forEach(onlyNeighboringShards::add));
+            onlyNeighboringShards.removeAll(this.loadedShards.keySet());
+            // For each of those shards, load the Atlas individually and find the relation and its
+            // members if it is there too.
+            final Set<Shard> neighboringShardsContainingRelation = new HashSet<>();
+            onlyNeighboringShards
+                    .forEach(shard -> this.policy.getAtlasFetcher().apply(shard).ifPresent(atlas ->
+                    {
+                        final Relation newRelation = atlas.relation(relation.getIdentifier());
```

Comment:
Can't you implement this already right now with the current `DynamicAtlasPolicy` definition? It seems like code block from line 821-840 could be implemented into `DynamicAtlasPolicy`'s `atlasFetcher` parameter.

Reply:
So this method here loads the very neighboring shards and looks at the relation members to see if any completely new members are present here or not. If yes, it adds the new shards to the list.

The previous code could load and expand on relations, but only if some members of the relation were partially outside of the current set of shards.

If you think I have misunderstood your question, can you send a link to the code that you think can replace this?

Reply:
@mertemin - please file a github issue if you see a refactor opportunity and we can address it. Otherwise, merging to make use of this functionality.